libvast/src/chunk.cpp

```diff
@@ -187,8 +187,8 @@ chunk::~chunk() noexcept {
   const auto* data = view_.data();
   const auto sz = view_.size();
   VAST_TRACEPOINT(chunk_delete, data, sz);
-  if (deleter_)
-    std::invoke(deleter_);
+  if (auto deleter = std::exchange(deleter_, {}))
+    std::invoke(std::move(deleter_));
```

Comment:
This looks fishy to me. Don't you want
```suggestion
  if (deleter_)
    std::invoke(std::exchange(deleter_, {}));
```
?

Reply:
Does the same thing except that it moves one time less, but fine by me.

Reply:
The difference is that your change moves out of `deleter_` both times.

Reply:
Ohhhh, I definitely did not want to move the member variable in the second one. 👀